homepage/models.py

```diff
@@ -69,6 +69,15 @@ class Req_Code(models.IntegerChoices):
     def __str__(self):
         return f'req_cour = {self.req_course_id}, desired_cour = {self.course_id}, req_code = {self.req_code}'

```

Comment:
I know this is not part of this PR but because you test it in line # 243 I pay attention to this string.
IMHO it will be more readable like:
```suggestion
        return f'Req course id: {self.req_course_id}, Desired course id: {self.course_id}, Req code: {self.req_code}'
```